### ISSUE TITLE
The layout is broken -- kind of

### ISSUE THREAD

**Issue by srameshr** [brh55/react-native-masonry#41]
Here is a screenshot: 

![screen shot 2017-09-07 at 10 48 56 pm](https://user-images.githubusercontent.com/8149408/30176092-d0c046aa-941e-11e7-89a3-53df56b9d9a6.png)

**Sometimes the columns are a miss**
![screen shot 2017-09-07 at 11 58 41 pm](https://user-images.githubusercontent.com/8149408/30179125-ef564b8c-9428-11e7-9c9c-68d0ba3f038f.png)

These quirk happen sometimes on navigation. When I navigate to and fro to this page.

In the below image, data is  ordered as 5,4,3,2,1. But the list renders them as below.
![screen shot 2017-09-08 at 12 08 13 am](https://user-images.githubusercontent.com/8149408/30179513-4c5267ca-942a-11e7-8660-a33a00311f7f.png)

```
<ScrollView style={{ flex: 1, padding: 5 }}>
        <Masonry
          bricks={this.state.contents}
          columns={2}
        />
      </ScrollView>
    );
```

**Comment by brh55** [comment]
Are the indexs not inserting into the right columns?

**Comment by brh55** [comment]
Doesn't currently account for height, this would be something that is added as a functionality in #5

**Comment by srameshr** [comment]
@brh55 Sometimes it does not insert into the right columns. Sometimes it does. This bug breaks the whole point to Masonry sometimes.

**Comment by brh55** [comment]
FlatList will not fix this issue, technically #38 should of fixed this

**Comment by srameshr** [comment]
@brh55 I am using the sorted option. But still this issue exists. All the screenshots i have posted here use the ```sorted``` option.

**Comment by zanisis** [comment]
yes my issue is same. sometimes image not showing . if i close page and come back again my image will be show. i using with react navigation

**Comment by srameshr** [comment]
@brh55 Any progress on this?

**Comment by brh55** [comment]
@srameshr Start by removing the scroll view wrapper and flex styles. Probably won't fix the issue, but you shouldn't need that anymore with the latest v0.4.0.

**Comment by brh55** [comment]
@srameshr is this happening with a nested view within a navigation component, or does it also happen when it is on it's own. I only ask because I don't see these issues with the updated demo).

**Comment by srameshr** [comment]
@brh55 Yes it happens with the nested view with react native navigation component. My current app does not have a scrollview wrapper.

**Comment by srameshr** [comment]
Nope! Its still the same. It takes 15 seconds to render just one image.

**Comment by srameshr** [comment]
@brh55 This issue still exists regardless of any changes to image size.

**Comment by brh55** [comment]
Yeah this is most likely with the rowHasChanged function, can you try to just return true by default. And see if this fixes the issue

**Comment by srameshr** [comment]
@brh55 No, its not working and there is a big problem. Since most of the time (not always) it renders only one column regardless of what the column size is, Apple is rejecting my build saying that the app is not optimised to take up the full screen. 

![screen shot 2017-10-04 at 10 41 44 am](https://user-images.githubusercontent.com/8149408/31160789-bcad98c0-a8f0-11e7-900c-5eb8581a5061.png)

This is the screenshot of the app that they sent.
![attachment-2600443384271620541screenshot-1003-164157 1](https://user-images.githubusercontent.com/8149408/31160798-cfd9dc38-a8f0-11e7-808b-01194ff4f59f.png)

Waiting for this fix, desperately.

**Comment by brh55** [comment]
Were you having these issues with v0.3.0 or after the flatlist changes?

**Comment by srameshr** [comment]
I don't remember! I dug through the code and I was not able to figure out exactly where and how the insert into column logic is written. Could you help me out with that, so that I can try to fix it.

**Comment by brh55** [comment]
Insert into column is at Masonry.js LOC 134. as well as assigning a column number at LOC 12.

**Comment by brh55** [comment]
@srameshr if you can email me, and perhaps give me access to your codebase (or a similarly set up codebase) it will make it significantly easier for me to narrow down the cause of this issue

**Comment by srameshr** [comment]
@brh55 Your email?

**Comment by brh55** [comment]
@srameshr  brandonhim@live.com

**Comment by srameshr** [comment]
@brh55 I dropped you a mail.

**Comment by vvavepacket** [comment]
hi guys do we have any temporary fixes for this? im encountering the same issue, the settimeout works sometimes but not always

**Comment by vvavepacket** [comment]
@srameshr  do u encounter this issue on both android and ios?

**Comment by srameshr** [comment]
```0.4.4``` fixes this.

**Comment by jacktator** [comment]
Experiencing the same issue with 0.4.7, wrapped in NativeBase 'Content' component.

**Comment by brh55** [comment]
@jacktator Are you experiencing uneven layouts? If so the current release of masonry only places them where they are initially assigned prior to resolving dimensions. This feature of best effort equally distributed layouts is available in the branch v0.5.0 but it not all additional planned updates have been finished so I haven't released it yet. Feel free to try and test and provide feedback.
